csharp/ql/src/semmle/code/csharp/dataflow/flowsources/AuthCookie.qll

```diff
@@ -0,0 +1,186 @@
+    m.getName() = "Configure" and
+    m.getDeclaringType().getName() = "Startup" and
+    m.getBody().getAChild+() = mc and
+    mc.getTarget() =
+      any(MicrosoftAspNetCoreBuilderCookiePolicyAppBuilderExtensions e).getUseCookiePolicyMethod() and
+    oc.getType() instanceof MicrosoftAspNetCoreBuilderCookiePolicyOptions and
+    getAValueForProp(oc, a, prop) = val and
+    result = val
+  )
+}
+
+/**
+ * A simplistic points-to alternative: given an object creation and a property name, get the values that property can be assigned.
+ *
+ * Assumptions:
+ *    - we don't reassign the variable that the creation is stored in
+ *    - we always access the creation through the same variable it is initially assigned to
+ *
+ * This should cover most typical patterns...
+ */
+Expr getAValueForProp(ObjectCreation create, Assignment a, string prop) {
+  // values set in object init
+  exists(MemberInitializer init, Expr src, PropertyAccess pa |
+    a.getLValue() = pa and
+    pa.getTarget().hasName(prop) and
+    init = create.getInitializer().(ObjectInitializer).getAMemberInitializer() and
+    init.getLValue() = pa and
+    DataFlow::localExprFlow(src, init.getRValue()) and
+    result = src
+  )
+  or
+  // values set on var that create is assigned to
+  exists(Expr src, PropertyAccess pa |
+    a.getLValue() = pa and
+    pa.getTarget().hasName(prop) and
+    DataFlow::localExprFlow(create, pa.getQualifier()) and
+    DataFlow::localExprFlow(src, a.getRValue()) and
+    result = src
+  )
+}
+
+/**
+ * Similar to getAValueForProp, checks if the given property was explicitly set to a value.
+ */
+predicate isPropertySet(ObjectCreation oc, string prop) {
+  exists(Assignment a, PropertyAccess pa |
+    a.getLValue() = pa and
+    pa.getTarget().hasName(prop) and
+    DataFlow::localExprFlow(oc, pa.getQualifier())
+  )
+  or
+  exists(Assignment a, PropertyAccess pa, Expr src |
+    a.getLValue() = pa and
+    pa.getTarget().hasName(prop) and
+    a.getRValue() = src and
+    DataFlow::localExprFlow(src,
+      oc.getInitializer().(ObjectInitializer).getAMemberInitializer().getRValue())
+  )
+}
+
+/**
+ * Tracks if a callback used in `OnAppendCookie` sets `Secure` to `true`.
+ */
+class OnAppendCookieSecureTrackingConfig extends OnAppendCookieTrackingConfig {
+  OnAppendCookieSecureTrackingConfig() { this = "OnAppendCookieSecureTrackingConfig" }
+
+  override string propertyName() { result = "Secure" }
+}
+
+/**
+ * Tracks if a callback used in `OnAppendCookie` sets `HttpOnly` to `true`.
+ */
+class OnAppendCookieHttpOnlyTrackingConfig extends OnAppendCookieTrackingConfig {
+  OnAppendCookieHttpOnlyTrackingConfig() { this = "OnAppendCookieHttpOnlyTrackingConfig" }
+
+  override string propertyName() { result = "HttpOnly" }
+}
+
+abstract class OnAppendCookieTrackingConfig extends DataFlow::Configuration {
+  bindingset[this]
+  OnAppendCookieTrackingConfig() { any() }
+
+  abstract string propertyName();
```

Comment:
Could you please add doc comments to the public entities: `OnAppendCookieTrackingConfig`, and `OnAppendCookieTrackingConfig::propertyName`?